kernel/vmlambda.ml

```diff
@@ -29,23 +32,52 @@ let is_value lc =
 let get_value lc =
   match lc with
   | Luint i -> val_of_uint i
-  | Lval v -> v
+  | Lval (_, v) -> v
   | Lint i -> val_of_int i
-  | _ -> raise Not_found
+  | _ -> assert false
+
+let hash_block tag args =
+  let open Hashset.Combine in
+  let fold accu v =
+    let h = match v with
+    | Luint i -> Uint63.hash i
```

Comment:
Should we also have a specific case for Lfloat? (well, according to float64_common.ml `Float64.hash` is exactly `Hashtbl.hash` so that sounds fine)

Reply:
Storing floats in structured values is asking for flat-float-array trouble, but maybe we can pull it off.